agent/engine/docker_task_engine.go

```diff
@@ -1133,15 +1133,23 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 	}
 
 	if execcmd.IsExecEnabledContainer(container) {
-		tId, err := task.GetID()
+		var herr error
+		tID, err := task.GetID()
 		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+			herr = &apierrors.HostConfigError{Msg: err.Error()}
```

Comment:
I don't think there's a need for this error to be `apierrors.HostConfigError` since we're not adding it to metadata anymore. Let's just use a plain onld `error`